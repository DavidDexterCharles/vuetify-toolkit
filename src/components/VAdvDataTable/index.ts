import VAdvDataTable from './VAdvDataTable'

export { VAdvDataTable }
export default VAdvDataTable
