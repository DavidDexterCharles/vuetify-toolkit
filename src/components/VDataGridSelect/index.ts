import VDataGridSelect from './VDataGridSelect'

export { VDataGridSelect }
export default VDataGridSelect
