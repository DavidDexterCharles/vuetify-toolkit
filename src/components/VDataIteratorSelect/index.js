import VDataIteratorSelect from './VDataIteratorSelect'

export { VDataIteratorSelect }
export default VDataIteratorSelect
