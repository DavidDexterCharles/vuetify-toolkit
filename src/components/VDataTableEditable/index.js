import VDataTableEditable from './VDataTableEditable'

export { VDataTableEditable }
export default VDataTableEditable
