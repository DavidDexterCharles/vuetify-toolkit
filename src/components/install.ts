import { VueConstructor, Component } from 'vue'

export interface VuetifyToolkitUseOptions {
  components?: Record<string, Component>
}

export function install (Vue: VueConstructor, args: VuetifyToolkitUseOptions) {
  if ((install as any).installed) return
  (install as any).installed = true
  const components = args.components || {}

  for (const key in components) {
    const component = components[key]
    if (component) {
      Vue.component(key, component as typeof Vue)
    }
  }
  // Used to avoid multiple mixins being setup
  // when in dev mode and hot module reload
  // https://github.com/vuejs/vue/issues/5089#issuecomment-284260111
  if ((Vue as any).$_vuetify_toolkit_installed) return
  (Vue as any).$_vuetify_toolkit_installed = true
}
