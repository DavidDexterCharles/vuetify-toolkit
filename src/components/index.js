import VTreeSelect from './VTreeSelect'
import VCascader from './VCascader'
import VDataGridSelect from './VDataGridSelect'
import VMdView from './VMdView'

import Vuetify from 'vuetify/lib'

const install = Vuetify.install

const components = {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}

Vuetify.install = (Vue, args) => {
  install.call(Vuetify, Vue, {
    components
  })
}

export default {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}

export {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VMdView
}
