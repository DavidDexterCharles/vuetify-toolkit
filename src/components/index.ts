import VTreeSelect from './VTreeSelect'
import VCascader from './VCascader'
import VDataGridSelect from './VDataGridSelect'
import VDateTimeSelect from './VDateTimeSelect'
import VMdView from './VMdView'
import VAdvDataTable from './VAdvDataTable'

export default {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VDateTimeSelect,
  VMdView,
  VAdvDataTable
}

export {
  VTreeSelect,
  VCascader,
  VDataGridSelect,
  VDateTimeSelect,
  VMdView,
  VAdvDataTable
}
