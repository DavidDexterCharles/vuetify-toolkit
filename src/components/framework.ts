import Vue from 'vue'
import { install } from './install'

export default class VuetifyToolkit {
  static install = install

  static installed = false

  public installed: string[] = []

  // Called on the new vuetify-toolkit instance
  // bootstrap in install beforeCreate
  // Exposes ssrContext if available
  init (root: Vue, ssrContext?: object) {}
}
