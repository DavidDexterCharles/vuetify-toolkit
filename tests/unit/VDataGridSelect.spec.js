// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify/lib'
// Utilities
import {
  mount,
  createLocalVue
} from '@vue/test-utils'
// component to be tested
import { VDataGridSelect } from '@/components'

Vue.use(Vuetify)

const vuetify = new Vuetify({
  icons: {
    iconfont: 'mdi'
  }
})

const localVue = createLocalVue()

const singleRootTwoChildren = [
  { id: 0, name: 'Root', children: [{ id: 1, name: 'Child' }, { id: 2, name: 'Child 2' }] }
]

const threeLevels = [
  { id: 0, name: 'Root', children: [{ id: 1, name: 'Child', children: [{ id: 2, name: 'Grandchild' }] }, { id: 3, name: 'Child' }] }
]

describe('VDataGridSelect.js', () => {
  it('renders', () => {
    const wrapper = mount(VDataGridSelect, {
      vuetify,
      propsData: {
        items: singleRootTwoChildren
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
