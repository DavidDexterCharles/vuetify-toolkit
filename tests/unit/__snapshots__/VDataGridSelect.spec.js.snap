// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDataGridSelect.js renders 1`] = `
<div class="v-input theme--light v-text-field v-select">
  <div class="v-input__control">
    <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-1" class="v-input__slot">
      <div class="v-select__slot">
        <div class="v-select__selections"><input id="input-1" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden">
      </div>
      <div class="v-menu">
        <div class="v-menu__content theme--light v-autocomplete__content" style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
          <div class="v-select-list v-card theme--light" id="list-1">
            <div class="v-data-table theme--light">
              <div class="v-data-table__wrapper">
                <table>
                  <colgroup>
                    <col class="">
                  </colgroup>
                  <thead class="v-data-table-header">
                    <tr>
                      <th role="columnheader" scope="col" aria-label="" aria-sort="none" class="text-start" style="width: 1px; min-width: 1px;"><span></span></th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr class="">
                      <td class="text-start">
                        <div class="v-data-table__checkbox v-simple-checkbox">
                          <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div class="v-data-footer">
                <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
                    <div class="v-input__control">
                      <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-13" class="v-input__slot">
                        <div class="v-select__slot">
                          <div class="v-select__selections">
                            <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="$vuetify.dataTable.itemsPerPageText" id="input-13" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                          </div>
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden" value="10">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="v-data-footer__pagination">1-1 of 1</div>
                <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
                <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
    </div>
  </div>
</div>
`;
